packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -244,7 +230,15 @@ export const Combobox = <T extends string | number>(props: ComboboxProps<T>) =>
     // Instead, stateReducer is called in the same tick as state changes, before that state is committed and rendered.
 
     onSelectedItemChange: ({ selectedItem }) => {
-      onChange(selectedItem);
+      if (selectedItem !== null || isClearable === true) {
+        // This hack is to allow a `null` value to be accepted by onChange.
+        // TypeScript can't infer from `isClearable === true` that the generic C type is `true`, but if it could,
+        // onChange would would allow `null` when it is true. The type works correctly outside of the component,
+        // where it's consumed, but not here. `selectedItem` could be cast to non-null, but that could accidentally
+        // allow it to be a different type that gets overwritten.
+        // If TS could infer here, then `onChange(selectedItem);` would pass type checking.
+        (onChange as (option: ComboboxOption<T> | null) => void)(selectedItem);
```

Comment:
is there a way to get TypeScript to recognize that onChange should allow `null` if `isClearable` is true?

Reply:
i feel like it should be possible via function overloading 🤔 but i can't quite get it to work